spansy/src/lib.rs

```diff
@@ -169,6 +169,16 @@ impl Span<str> {
     }
 }
 
+impl AsRef<Span<[u8]>> for Span<str> {
+    fn as_ref(&self) -> &Span<[u8]> {
+        // # Safety
+        // Span<str> can be safely converted to Span<[u8]> because they have
```

Comment:
Is there a guarantee that the `Span` struct is stored in contiguous memory? e.g. what if its `data` field and `indices` field are stored non-contiguously?